### PR TITLE
[ALLUXIO-3044] Add wasb as a top-level UFS

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-3044

### CODE REVIEW

File: underfs/wasb/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the “License”). You may not use this work except in compliance with the License, which is
```
Comment (calvinjia) [inline underfs/wasb/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory]: The `"` is not consistent?

File: underfs/wasb/src/test/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the “License”). You may not use this work except in compliance with the License, which is
```
Comment (calvinjia) [inline underfs/wasb/src/test/resources/log4j.properties]: The `"` is not consistent?

File: underfs/wasb/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory (hunk 1 above)
Comment (apc999) [reply underfs/wasb/src/main/resources/META-INF/services/alluxio.underfs.UnderFileSystemFactory]: fixed

File: underfs/wasb/src/test/resources/log4j.properties (hunk 2 above)
Comment (apc999) [reply underfs/wasb/src/test/resources/log4j.properties]: fixed

**Comment by apc999** [comment]
@gpang  @calvinjia  PTAL

File: underfs/hdfs/pom.xml [hunk 3]
```diff
@@ -159,25 +159,20 @@
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-dependency-plugin</artifactId>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <version>1.8</version>
```
Comment (calvinjia) [inline underfs/hdfs/pom.xml]: Add to plugin management?

File: underfs/hdfs/pom.xml [hunk 4]
```diff
@@ -159,25 +159,20 @@
...
-                  <groupId>${project.groupId}</groupId>
-                  <artifactId>${project.artifactId}</artifactId>
-                  <version>${project.version}</version>
-                  <type>${project.packaging}</type>
-                  <overWrite>true</overWrite>
-                </artifactItem>
-              </artifactItems>
-              <outputDirectory>${lib.path}</outputDirectory>
+              <target>
+                <copy todir="${basedir}/../../lib" overwrite="true"
```
Comment (calvinjia) [inline underfs/hdfs/pom.xml]: Cannot do something like `parent.parent.basedir`?

File: underfs/wasb/pom.xml [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    <groupId>org.alluxio</groupId>
+    <artifactId>alluxio-underfs</artifactId>
+    <version>1.7.0-SNAPSHOT</version>
+  </parent>
+  <artifactId>alluxio-underfs-wasb</artifactId>
+  <name>Alluxio Under File System - Microsoft Azure Blob Storage</name>
+  <description>Microsoft Azure Blob Storage Under File System implementation</description>
+
+  <properties>
+    <ufs.hadoop.version>2.7.3.2.6.1.0-129</ufs.hadoop.version>
```
Comment (calvinjia) [inline underfs/wasb/pom.xml]: should this have `wasb` somewhere?

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * An {@link UnderFileSystem} uses the Microsoft Azure Blob Storage.
+ */
+@ThreadSafe
+public final class WasbUnderFileSystem extends HdfsUnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(WasbUnderFileSystem.class);
+
+  /**
+   * Constant for the wasb URI scheme.
+   */
```
Comment (calvinjia) [inline underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: `/** */` format?

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  @Override
+  public long getBlockSizeByte(String path) throws IOException {
+    // wasb is an object store, so use the default block size, like other object stores.
+    return alluxio.Configuration.getBytes(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT);
+  }
+
+  // Not supported
+  @Override
+  public List<String> getFileLocations(String path) throws IOException {
+    LOG.debug("getFileLocations is not supported when using default ObjectUnderFileSystem.");
```
Comment (calvinjia) [inline underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: `when using wasb`?

File: underfs/hdfs/pom.xml (hunk 3 above)
Comment (apc999) [reply underfs/hdfs/pom.xml]: done

File: underfs/hdfs/pom.xml (hunk 4 above)
Comment (apc999) [reply underfs/hdfs/pom.xml]: unfortunately, `parent.parent.basedir` is not supported by `antrun`

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java (hunk 6 above)
Comment (apc999) [reply underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: fixed

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java (hunk 7 above)
Comment (apc999) [reply underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: fixed

File: underfs/wasb/pom.xml (hunk 5 above)
Comment (apc999) [reply underfs/wasb/pom.xml]: changed to `wasb.hadoop.version`

**Pull request merged**